Api/Api.cs

```diff
@@ -358,6 +432,28 @@ public virtual IEnumerable<MarketHoursSegment> MarketToday(DateTime time, Symbol
         }
 
         /// <summary>
+        /// Creates the path for the downloaded data to be saved in the same folder heirarchy as Lean
+        /// </summary>
+        /// <param name="symbol">Symbol of security of which data will be requested.</param>
+        /// <param name="resolution">Resolution of data requested.</param>
+        /// <param name="date">Date of the data requested.</param>
+        /// <returns>Path that specifies where data will be downloaded</returns>
+        private string CreatePathForDownloadedData(Symbol symbol, Resolution resolution, DateTime date)
+        {
+            string path;
+
+            if (resolution == Resolution.Daily || resolution == Resolution.Hour)
+                path = Config.Get("data-folder") + symbol.ID.SecurityType.ToLower() + "/" + symbol.ID.Market.ToLower() +
+                       "/" + resolution.ToLower() + "/" + symbol.Value.ToLower() + ".zip";
+            else
+                path = Config.Get("data-folder") + symbol.ID.SecurityType.ToLower() + "/" + symbol.ID.Market.ToLower() +
```

Comment:
Please pass in a "data-folder" variable in the constructor, and save it to the class, so its not loading from config each time.